src/WingetCreateCLI/PromptHelper.cs

```diff
@@ -0,0 +1,270 @@
+            }
+            else if ((elementType = Nullable.GetUnderlyingType(propertyType)) != null)
+            {
+                if (elementType.IsEnum)
+                {
+                    PromptEnum(model, property, elementType, false, message: message);
+                }
+            }
+            else if (propertyType.IsList())
+            {
+                elementType = propertyType.GetGenericArguments().SingleOrDefault();
+                if (elementType == typeof(string) || typeof(IEnumerable<string>).IsAssignableFrom(propertyType))
+                {
+                    var value = Prompt.List<string>(message, minimum: minimum, validators: new[] { FieldValidation.ValidateProperty(validationModel, validationName, defaultValue) });
+                    if (!value.Any())
+                    {
+                        value = null;
+                    }
+
+                    property.SetValue(model, value);
+                }
+                else if (elementType == typeof(int))
+                {
+                    // The only field that takes in List<int> is InstallerSuccessCodes, which only has constraints on the number of entries.
+                    // TODO: Add validation for checking the number of entries in the list.
+                    var value = Prompt.List<int>(message, minimum: minimum);
+                    property.SetValue(model, value);
+                }
+                else if (elementType.IsEnum)
+                {
+                    var value = Prompt.MultiSelect(message, Enum.GetNames(elementType), minimum: 0);
+
+                    if (value.Any())
+                    {
+                        Type genericListType = typeof(List<>).MakeGenericType(elementType);
+                        var enumList = (IList)Activator.CreateInstance(genericListType);
+
+                        foreach (var item in value)
+                        {
+                            var enumItem = Enum.Parse(elementType, item);
+                            enumList.Add(enumItem);
+                        }
+
+                        property.SetValue(model, enumList);
+                    }
+                }
+                else if (elementType == typeof(PackageDependencies))
+                {
+                    List<PackageDependencies> packageDependencies = (List<PackageDependencies>)property.GetValue(model) ?? new List<PackageDependencies>();
+
+                    PromptForPackageDependencies(packageDependencies);
+                    if (packageDependencies.Any())
+                    {
+                        property.SetValue(model, packageDependencies);
+                    }
+                    else
+                    {
+                        property.SetValue(model, null);
+                    }
+                }
+            }
+            else
+            {
+                if (property.PropertyType == typeof(InstallerSwitches))
```

Comment:
Asking same question, can this be made generic?

Reply:
Made generic